File: Java/Sorting/SelectionSort/SelectionSort.java
class SelectionSort{
    public static void main(String[] args){
        int[] unsorted = new int[]{30,20,40,80,220,6969};
	    int hold;
        for (int i = 0; i<unsorted.length; i++){
            int min = unsorted[i];
            int minLoc = i;

            for (int j = i; j<unsorted.length; j++){
                if (min > unsorted[j]){
                    min = unsorted[j];
                    minLoc = j;
                }
            }
            hold = unsorted[i];
            unsorted[i] = min;
            unsorted[minLoc] = hold;            
        }
        for (int val: unsorted){
		System.out.print(val + " ");
	}
	System.out.print("\n");
    }
}

File: TypeScript/SelectionSort/SelectionSort.ts
export default function SelectionSort(nums :number[]) :number[]{
    for (let i = 0 ; i< nums.length; i++){ //finish with a sorted element
        let min = nums[i];
        let minindex = i;
        for (let j  = i; j<nums.length; j++ ){
            if (nums[j] < min){
                min = nums[j];
                minindex = j;
            }
        }
            nums[minindex] = nums[i]; //nums[1] = 3
            nums[i] = min;             //nums[0] = 1
    }
    return nums;
}
console.log(SelectionSort([3,4,12,-21,222,1,2]));